### PR TITLE
[docs] Clarify product split

### LINKED ISSUE #1078: The license instructions are confusing in EULA vs the repository's README - need clarification
Goal:
Trying to see if using data grid from material-ui-x for production is legally allowed or not.

Groundwork:
In store, it shows the license as EULA: 
https://material-ui.com/store/items/material-ui-x/
https://docs.google.com/document/d/1nkd5JP5wCefo6UwpVi6PnbTaw8DxhstkCZlfHV5ZgfI/edit?usp=sharing

But in readme of this repo, it shows datagrid - MIT license and x-grid - Commercial license, 

Query:
so when i import material-ui-x - does it mean i am allowed to use Data-grid in production environment of my project (saleable to customers)
OR 
are they available freely for development and learning purposes only.
Can someone help clarify/confirm urgently?

Raised as issue, since new users to license agreement's complex terms the readme kind of is throwing me in doubt. So after above answer, can somone also update the readme or else once i get an answer i can try to raise a PR to update the same for clarity to developers/consmers.

Thanks in advance.

### PR DESCRIPTION
**Pull request by oliviertassinari**
Closes #1078

Interesting perspective from WordPress's founder: https://youtu.be/49HSZ4JQlHY?t=285

### CODE REVIEW

File: docs/src/pages/components/data-grid/columns/columns.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ In addition, `flex` does not work if the combined width of the columns that have
 
 {{"demo": "pages/components/data-grid/columns/ColumnFluidWidthGrid.js", "bg": "inline"}}
 
-## Column resizing [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-x/)
+## Column resizing [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-enterprise/)
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/columns/columns.md:68]: We will point the /pricing page in the future, this is temporary

File: docs/src/pages/components/data-grid/columns/columns.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ In addition, `flex` does not work if the combined width of the columns that have
 
 {{"demo": "pages/components/data-grid/columns/ColumnFluidWidthGrid.js", "bg": "inline"}}
 
-## Column resizing [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-x/)
+## Column resizing [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-pro/)
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/columns/columns.md:68]: It's temporary. It will point to the `/pricing` page in the future.

**Comment by oliviertassinari** [comment]
@Sd100 Does it clarify it?

File: README.md [hunk 3]
```diff
@@ -4,12 +4,24 @@
...
 <h1 align="center">Material-UI X</h1>
 
-The Material-UI X mono-repo gathers Material-UI advanced components.
+The [main repository](https://github.com/mui-org/material-ui) of Material-UI focuses on providing a ready-to-use design system, as well as empowering developers to build their own.
+
+This Material-UI X mono-repo gathers the Material-UI advanced components. It's an extension of the core. The most fundamental features are MIT licensed and the most advanced ones commercial.
+
+## How do we decide if a feature is MIT or commercial?
+
+We have been building MIT React components since 2014. we have learned a lot about the strength and weaknesses of the MIT license model. We believe that any open source project will ultimately become the dominant and best solution in any market it enters. The **ultimately** aspect is very important. While for the simple components, we have reached this point a long time ago. We believe that advanced components won't reach this point before multiple decades. So what should we do? Wait? No! We are using a commercial license to sustain the development of the most advanced features, where the open-source model can't.
```
Comment (oliviertassinari) [inline README.md:13]: *ultimately* means that commercial components can't be our end-goal, hence the need for a low-code shift going forward.

Comment (oliviertassinari) [reply README.md:13]: I have first hard that from WordPress's founder: https://youtu.be/49HSZ4JQlHY?t=285

File: README.md [hunk 4]
```diff
@@ -4,12 +4,30 @@
 
 <h1 align="center">Material-UI X</h1>
 
-The Material-UI X mono-repo gathers Material-UI advanced components.
+The [main repository](https://github.com/mui-org/material-ui) of Material-UI focuses on providing a ready-to-use design system, as well as empowering developers to build their own.
+
+This Material-UI X mono-repo gathers the Material-UI advanced components. It's an extension of the core. The most fundamental features are MIT licensed and the most advanced ones commercial.
```
Comment (mbrookes) [inline README.md:9]:
```suggestion
This Material-UI X mono-repo gathers the advanced Material-UI components – it's an extension of the core. The fundamental features are MIT licensed, and the more advanced ones commercial.
```

File: README.md [hunk 5]
```diff
@@ -4,12 +4,30 @@
...
+
+## How do we decide if a feature is MIT or commercial?
+
+We have been building MIT React components since 2014.
```
Comment (mbrookes) [inline README.md:13]:
```suggestion
We have been building MIT React components since 2014,
```

File: README.md [hunk 6]
```diff
@@ -4,12 +4,30 @@
...
+We have learned a lot about the strength and weaknesses of the MIT license model.
```
Comment (mbrookes) [inline README.md:14]:
```suggestion
and have learned much about the strengths and weaknesses of the MIT license model.
```

File: README.md [hunk 7]
```diff
@@ -4,12 +4,30 @@
...
+The health of this model is improving every day. As the community grows, it increases the probability that developers contribute improvements to the project.
+However, we believe that have reached the sustainability limits of what the model can support for advancing our mission forward. We have seen too many MIT components moving slowly or getting abandoned. The community isn't contributing improvements as fast as the problems deserved to be solved.
```
Comment (mbrookes) [inline README.md:16]:
```suggestion
However, we believe that we have reached the sustainability limits of what the model can support for advancing our mission forward. We have seen too many MIT licensed components moving slowly or getting abandoned. The community isn't contributing improvements as fast as the problems deserved to be solved.
```

"too many MIT licensed components"? (i.e external to Material-UI?)

File: README.md [hunk 8]
```diff
@@ -4,12 +4,30 @@
...
+
+What should we do? Wait? No! We are using a commercial license to forward the development of the most advanced features, where the MIT model can't.
```
Comment (mbrookes) [inline README.md:18]:
```suggestion
We are using a commercial license to forward the development of the most advanced features, where the MIT model can't sustain it.
```

File: README.md [hunk 9]
```diff
@@ -4,12 +4,30 @@
...
+A solution to a problem should only be commercial if it has no MIT alternatives.
+
+We provide three options:
+
+- Community. This plan contains the MIT components that are sustainable by the contributions of the open-source community. Free and forever.
```
Comment (mbrookes) [inline README.md:23]:
```suggestion
- **Community**. This plan contains the MIT components that are sustainable by the contributions of the open-source community. Free forever.
```

File: README.md [hunk 10]
```diff
@@ -4,12 +4,30 @@
...
+- Pro. This plan contains the features that are at the limit of what the open-source model can sustain. For instance, providing a very comprehensive set of components. From a price perspective, the plan is designed to be accessible by most professionals.
```
Comment (mbrookes) [inline README.md:24]:
```suggestion
- **Pro**. This plan contains the features that are at the limit of what the open-source model can sustain. For instance, providing a very comprehensive set of components. From a price perspective, the plan is designed to be accessible to most professionals.
```

File: README.md [hunk 11]
```diff
@@ -4,12 +4,30 @@
...
+- Premium. This plan contains the most advanced features.
```
Comment (mbrookes) [inline README.md:25]:
```suggestion
- **Premium**. This plan contains the most advanced features.
```

File: README.md [hunk 12]
```diff
@@ -4,12 +4,30 @@
...
 
 ## Components
 
-- [DataGrid](https://material-ui.com/components/data-grid/) MIT
-- [XGrid](https://material-ui.com/components/data-grid/#commercial-version) Commercial license
+- [DataGrid](https://material-ui.com/components/data-grid/), publised under `@material-ui/data-grid`, MIT licensed.
+- [XGrid](https://material-ui.com/components/data-grid/#commercial-version), publised under `@material-ui/x-grid`, Commercial license.
```
Comment (mbrookes) [inline README.md:30]:
```suggestion
- [XGrid](https://material-ui.com/components/data-grid/#commercial-version), publised under `@material-ui/x-grid`, commercially licensed.
```

**Review by mbrookes — CHANGES_REQUESTED** [review]

**Review by mbrookes — APPROVED** [review]

**Review by dtassone — APPROVED** [review]

**Review by DanailH — APPROVED** [review]

**Pull request merged**